### DISCUSSION TITLE
My dev branch is one more commit behind every pull request merge

### DISCUSSION THREAD

**Discussion by bubycloudr** [community/community#23353] [New to GitHub]
<p>I have a dev branch, a staging branch and temporary feature branches. Every time I merge a pull request from dev to staging the dev seems to be one more commit behind? I don’t understand whats happening. How can I fix this?<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/3/c315db43c28378f2d0d6c3a631a71867af882b1c.png" data-download-href="/uploads/short-url/rPNVUB5d624t2SNgxlcix8qMs9u.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/60745997/181076484-0ec0bf18-369e-4843-ad05-c8e368227538.png" alt="image" data-base62-sha1="rPNVUB5d624t2SNgxlcix8qMs9u" width="690" height="47" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/3/c315db43c28378f2d0d6c3a631a71867af882b1c_2_690x47.png, https://aws1.discourse-cdn.com/github/original/3X/c/3/c315db43c28378f2d0d6c3a631a71867af882b1c.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/3/c315db43c28378f2d0d6c3a631a71867af882b1c.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/3/c315db43c28378f2d0d6c3a631a71867af882b1c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">908×62 5.43 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<p>You’ll have to check the history to be sure, but I assume that’s a merge commit created when you merge. :slightly_smiling_face:</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="TerThesz" data-post="1" data-topic="192829">
<div class="title">
<div class="quote-controls"></div>
 TerThesz:</div>
<blockquote>
<p>Every time I merge a pull request from dev to staging the dev seems to be one more commit behind?</p>
</blockquote>
</aside>
<p>The screenshot message actually says 1 commit ahead and <em>22 behind</em>.</p>
<p>There seem to be some serious synching problems between those two branches. Also, it’s not clear if this message refers to your local clone status or the status of the repository on GitHub.</p>
<p>As usual, links to actual repositories would help helper to help out.</p>

**Comment by bubycloudr** [comment]
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/TerThesz/your-http">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/TerThesz/your-http" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/60745997/181076489-433fe796-918e-4c88-989f-796da52ef64c.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/d/2d63ae5d81b4e9cc2da953b97df455471670e384_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/2/d/2d63ae5d81b4e9cc2da953b97df455471670e384_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/2/d/2d63ae5d81b4e9cc2da953b97df455471670e384.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/d/2d63ae5d81b4e9cc2da953b97df455471670e384_2_10x10.png"></div>

<h3><a href="https://github.com/TerThesz/your-http" target="_blank" rel="noopener nofollow ugc">GitHub - TerThesz/your-http: Simple http server made for fun.</a></h3>

  <p>Simple http server made for fun. Contribute to TerThesz/your-http development by creating an account on GitHub.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>The dev branch is 2 commits ahead because I am adding new features (I will make a pull request when I finish) and its 32 commits behinds because I have merged 32 pull requests (Well I have made more pull requests but I accidentally deleted the dev branch so I had to make a new one)</p>

**Comment by tajmone** [comment, marked as answer]
<aside class="quote no-group" data-username="TerThesz" data-post="1" data-topic="192829">
<div class="title">
<div class="quote-controls"></div>
 TerThesz:</div>
<blockquote>
<p>Every time I merge a pull request from dev to staging the dev seems to be one more commit behind?</p>
</blockquote>
</aside>
<p>This might depend on the merge strategy you’re using. By default, GitHub will  create a merge commit for every PR merge (`–ff``), which is probably why you’re seeing the extra commit. I prefer to always use the Rebase-Merge strategy, which preserves a linear history.</p>
<p>You can change this in your repository settings, and decide which should be the default PR merging strategy. You can also enforce linear history in branch protection rules, so you don’t accidentally introduce those extra merge-commits.</p>
<p><a href="https://www.bogotobogo.com/cplusplus/Git/Git_GitHub_Fast-Forward_Merge.php" class="onebox" target="_blank" rel="noopener nofollow ugc">https://www.bogotobogo.com/cplusplus/Git/Git_GitHub_Fast-Forward_Merge.php</a></p>
